v1/plugin/stream_collector_proxy.go

```diff
@@ -199,9 +219,13 @@ func (p *StreamProxy) setMaxMetricsBuffer(i int64) {
 	p.maxMetricsBuffer = i
 }
 
-func sendReply(metrics []*rpc.Metric, stream rpc.StreamCollector_StreamMetricsServer) {
+func sendReply(taskid string, metrics []*rpc.Metric, stream rpc.StreamCollector_StreamMetricsServer) {
 	if len(metrics) == 0 {
-		fmt.Println("No metrics available to send")
+		log.WithFields(
+			log.Fields{
+				"_block": "sendReply",
```

Comment:
The taskid should be included in these fields.  Further if you wanted to potentially simplify this you could create a logger with the common fields at the top and use it below reducing some of the duplication in this method.